### PR TITLE
Feature/che 565 refactor flows

### PR DESCRIPTION
**Pull request by EvansPie**
CHE-565

# What this PR does

Refactors, removes and adds payment flows

# Notable decisions & other stuff

None

# Before merging

_QA_

- [x] I manually tested this with a demo application (simulator)
- [ ] I manually tested this with a demo application (real device)
- [ ] I wrote unit tests for each new util-like function
- [ ] I wrote at least basic e2e tests for the new functionalities or new paths
- [ ] If this PR introduces UI changes, I manually tested this PR on iPhone 8 (iOS 10, 12, 14), iPhone 12, and iPhone 12 Max
- [ ] If this PR modifies the behavior of an API, I did my best to make my new API backward compatible
- [ ] If this PR modifies the API, I attempted to implement the feature with an example

_Documentation_

- [ ] If this PR modifies the API, I updated the API Reference.
- [ ] If this PR adds new options to the API, I wrote some documentation or guide in the online documentation
- [ ] If the PR modifies the API, I wrote some documentation to deprecate it

# After merging

- Make sure a new release has been pushed to Cocoapods trunk
- 
Simulator Screen Shot - iPhone 12 - 2021-05-13 at 13 00 45
Simulator Screen Shot - iPhone 12 - 2021-05-13 at 13 00 30
Simulator Screen Shot - iPhone 12 - 2021-05-13 at 14 13 22

### CODE REVIEW

File: Sources/PrimerSDK/Classes/User Interface/Confirm Mandate/ConfirmMandateViewModel.swift
Comment (ceriksson) [inline Sources/PrimerSDK/Classes/User Interface/Confirm Mandate/ConfirmMandateViewModel.swift:109]: True, a direct debit token cannot be used to auth/capture a payment. However, it might still make sense to pass it back to the merchant as they might want to store the token or something? Wdyt?

**Review by ceriksson — COMMENTED** [review]
Looks good, let's align on comments then merge :)

File: Sources/PrimerSDK/Classes/Data Models/PrimerSessionFlow.swift
Comment (xevious78) [inline Sources/PrimerSDK/Classes/Data Models/PrimerSessionFlow.swift:39]: [Feedback]
We can simplify the naming of `primerWithVault` and `primerCheckout` - having `primer` as part of the name doesn't give that much information about what this does.
I think we can just call `vault` and `checkout`.

**Review by xevious78 — CHANGES_REQUESTED** [review]
Good first step but I think we should go further 💪 

This is the kind of UI the developer shouldn't be able to create: the user is asked to *vault* a payment method for a *payment*. 
CleanShot 2021-05-03 at 18 36 27@2x

From what I see, there's a bit of confusion between the "vault" flow (using the checkout to vault a payment method) and the "vault" manager (the UI that allows quick payment). 
What I suggest is:
- `primerWithVault` renamed `vault` that enables the user to vault a payment method among a list of payment methods. The dev shouldn't be able to show the vault manager & the amount.
- `primerCheckout` renamed `checkout` that enables the user to pay with a payment method, including a vaulted one. This is the Universal Checkout. Hiding the vault manager should be a setting of the flow, and not a flow in and on itself.

@EvansPie @ceriksson what do you think?

File: Sources/PrimerSDK/Classes/Data Models/PrimerSessionFlow.swift
Comment (EvansPie) [reply Sources/PrimerSDK/Classes/Data Models/PrimerSessionFlow.swift:39]: Done

File: Sources/PrimerSDK/Classes/Data Models/PrimerSessionFlow.swift
Comment (EvansPie) [reply Sources/PrimerSDK/Classes/Data Models/PrimerSessionFlow.swift]: Done

File: Sources/PrimerSDK/Classes/User Interface/Confirm Mandate/ConfirmMandateViewModel.swift
Comment (EvansPie) [reply Sources/PrimerSDK/Classes/User Interface/Confirm Mandate/ConfirmMandateViewModel.swift:109]: Makes sense, fixed!

File: Sources/PrimerSDK/Classes/Data Models/PrimerSessionFlow.swift
Comment (EvansPie) [reply Sources/PrimerSDK/Classes/Data Models/PrimerSessionFlow.swift]: @xevious78 I agree on what you say above. I believe the above screenshot is an outcome of not checking the flow and just adding the amount if the developer has provided one. We have to change this. I'll create a ticket for this one.

Comment (xevious78) [reply Sources/PrimerSDK/Classes/Data Models/PrimerSessionFlow.swift]: @EvansPie I can pick this up and make sure that the flows are properly represented. Let's have a quick sync on this :)

**Review by ceriksson — COMMENTED** [review]
Will let Aladin approve this once changes have been made :)

File: Sources/PrimerSDK/Classes/Data Models/PrimerInternalSessionFlow.swift
Comment (ceriksson) [inline Sources/PrimerSDK/Classes/Data Models/PrimerInternalSessionFlow.swift:72]: is this so we dont expose the singleton to the developer?

**Review by ceriksson — COMMENTED** [review]
lgtm, just waiting for the screenshots, then I'll approve

File: Sources/PrimerSDK/Classes/Data Models/PrimerInternalSessionFlow.swift
Comment (EvansPie) [reply Sources/PrimerSDK/Classes/Data Models/PrimerInternalSessionFlow.swift:72]: This isn't a singleton, just an enum. This is so we can use naming that makes more sense internally, while not making breaking changes.

**Review by ceriksson — APPROVED** [review]
lgtm 👍

**Review by xevious78 — APPROVED** [review]

**Pull request merged**
